src/libswid.cpp

```diff
@@ -76,20 +76,27 @@ unsigned int swid_root_get_type(SWIDHandle data) {
  */
 
 SWIDIOHandle swid_create_io(const char * backend) {
-	return (SWIDIOHandle)get_swidtagio(backend);
+	SWIDIOHandle io = new SWIDTagIO();
+	try {
+		io->setBackend(backend);
+	} catch (...) {
```

Comment:
this hides exceptions without reporting them anywhere which is an anti-pattern. this will hide bugs. I get it that C doesn't allow exceptions but this should log the info in the exception IMO.

Reply:
Again, this is something which didn't get here due to the small scope.